src/utils/size-observer.js

```diff
@@ -14,26 +14,96 @@
  * limitations under the License.
  */
 
-import {pushIfNotExist, removeItem} from './array';
+import {remove} from './array';
 import {rethrowAsync} from '../log';
 
 /** @typedef {function(../layout-rect.LayoutSizeDef)} */
-let ContentSizeObserverCallbackDef;
+let SizeObserverCallbackDef;
+
+/** @enum {number} */
+const Type = {
+  CONTENT: 0,
+  BORDER_BOX: 1,
+};
 
 /** @const {!WeakMap<!Window, !ResizeObserver>} */
 const observers = new WeakMap();
 
-/** @const {!WeakMap<!Element, !Array<!ContentSizeObserverCallbackDef>>} */
+/** @const {!WeakMap<!Element, !Array<{type: Type, callback: !SizeObserverCallbackDef}>>} */
 const targetObserverMultimap = new WeakMap();
 
-/** @const {!WeakMap<!Element, !../layout-rect.LayoutSizeDef>} */
-const targetSizeMap = new WeakMap();
+/** @const {!WeakMap<!Element, !ResizeObserverEntry>} */
```

Comment:
Do we have a use-case for ever looking at `contentSize`? (vs. always using borderBox)

Reply:
All other use cases should normally be `contentSize`. Partially why we were so excited about the `ResizeObserver`. This is because a custom element's size itself is not good for knowing how the content will be laid out. As an example, the old `amp-fit-text` computes the font-size a bit incorrectly because it can only rely on the border box. So, if such an element has no border/padding - we're all good. If it has a small border/padding - things are also usually ok. But if it has a large border/padding - we're completely off. The new `amp-fit-text` doesn't have that problem because we're using content-box size. This is true for most of other cases. The pausing is a bit unique here.